### PR TITLE
use implicit padding when possible

### PR DESCRIPTION
**Pull request by ngimel**

### CODE REVIEW

**Comment by soumith** [comment]
please ignore cla bot, it's a false-positive

File: fairseq/models/fconv.py [hunk 1]
```diff
@@ -121,10 +125,14 @@ def forward(self, src_tokens, src_lengths):
         for proj, conv in zip(self.projections, self.convolutions):
             residual = x if proj is None else proj(x)
             x = F.dropout(x, p=self.dropout, training=self.training)
-            padding_l = (conv.kernel_size[0] - 1) // 2
-            padding_r = conv.kernel_size[0] // 2
-            x = F.pad(x, (0, 0, 0, 0, padding_l, padding_r))
-            x = conv(x)
+            if conv.kernel_size[0] % 2 == 1:
+#padding is implicit in the conv
```
Comment (myleott) [inline fairseq/models/fconv.py]: nit: please indent the comment to align with `x = conv(x)` below

File: fairseq/models/fconv.py [hunk 2]
```diff
@@ -98,9 +98,13 @@ def __init__(self, dictionary, embed_dim=512, max_positions=1024,
...
             self.projections.append(Linear(in_channels, out_channels)
                                     if in_channels != out_channels else None)
+            if kernel_size % 2 == 1:
+               padding = kernel_size //2
+            else:
+               padding = 0
             self.convolutions.append(
                 ConvTBC(in_channels, out_channels * 2, kernel_size,
-                        dropout=dropout)
+                        dropout=dropout, padding = padding)
```
Comment (myleott) [inline fairseq/models/fconv.py]: nit: please remove space around operator, i.e., `padding=padding`.
